tests/core/test_encode_typed_data/test_encode_typed_data_full_messages.py

```diff
@@ -0,0 +1,69 @@
+import json
+import os
+
+from eth_utils import (
+    ValidationError,
+)
+import pytest
+
+from eth_account.messages import (
+    encode_typed_data,
+)
+
+loaded_files = {}
+for root, _, files in os.walk("tests/eip712_messages"):
+    for file in files:
+        if file.endswith(".json"):
+            folder_name = os.path.basename(root)
+            variable_name = f"{folder_name}_{os.path.splitext(file)[0]}"
+
+            with open(os.path.join(root, file), "r") as f:
+                loaded_files[variable_name] = json.load(f)
+
+valid = [message for message in loaded_files if message.startswith("valid")]
+invalid = [message for message in loaded_files if message.startswith("invalid")]
+one_arg_invalid = [
+    message for message in loaded_files if message.startswith("one_arg_invalid")
+]
+
+
+def convert_to_3_arg(message):
+    domain_data = message["domain"]
+    message_types = message["types"]
+    if message_types.get("EIP712Domain"):
+        message_types.pop("EIP712Domain")
+    message_data = message["message"]
+    return domain_data, message_types, message_data
+
+
+def test_there_are_no_duplicate_test_cases():
+    string_test_cases = [
+        json.dumps(msg, sort_keys=True) for msg in loaded_files.values()
+    ]
+    assert len(string_test_cases) == len(set(string_test_cases))
+
+
+@pytest.mark.parametrize("message", valid)
+def test_valid_messages(message):
+    assert encode_typed_data(full_message=loaded_files[message]) == encode_typed_data(
+        *convert_to_3_arg(loaded_files[message])
+    )
+
+
+@pytest.mark.parametrize("message", invalid)
+def test_invalid_messages(message):
+    with pytest.raises(ValueError):
+        encode_typed_data(full_message=loaded_files[message])
+
+    with pytest.raises(ValueError):
+        encode_typed_data(*convert_to_3_arg(loaded_files[message]))
+
+
+@pytest.mark.parametrize("message", one_arg_invalid)
+def test_messages_that_are_only_invalid_for_one_arg_encoding(message):
+    with pytest.raises(ValidationError):
+        encode_typed_data(full_message=loaded_files[message])
+    try:
```

Comment:
Curious why the `try` / `except` here? If the call without `full_message` fails, it will fail the test. Maybe leave an inline comment in the test if we want to clarify why it shouldn't throw an exception.

Reply:
Noop, just being a little overly-explicit I think. changed!